### PR TITLE
[NETBEANS-6361] Code completion for class fields doesn't work.

### PR DESCRIPTION
**Pull request by ppisl**
This fixing resolving types of fields.

### CODE REVIEW

File: groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/completion/inference/TypeInferenceVisitor.java [hunk 1]
```diff
@@ -99,6 +99,8 @@ public void visitField(FieldNode node) {
         if (sameVariableName(leaf, node)) {
             if (node.hasInitialExpression()){
                 guessedType = deriveExpressonType(node.getInitialExpression());
+            } else if (node.getType() != null) {
+                    guessedType = node.getType();
```
Comment (sdedic) [inline groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/completion/inference/TypeInferenceVisitor.java:103]: Q: would `GroovyUtils.findInferredType()` in addition to getType() bring some benefits here ?

File: groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/occurrences/TypeVisitor.java [hunk 2]
```diff
@@ -139,6 +139,11 @@ public void collect() {
                         visitClosureListExpression((ClosureListExpression) scope);
                         return;
                     }
+                } else if (scope instanceof ClassNode) {
+                    ClassNode classNode = (ClassNode) scope;
+                    for (FieldNode fieldNode: classNode.getFields()) {
+                        visitField(fieldNode);
```
Comment (sdedic) [inline groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/occurrences/TypeVisitor.java:145]: just fields ?

**Review by sdedic — APPROVED** [review]

File: groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/completion/inference/TypeInferenceVisitor.java (hunk 1 above)
Comment (ppisl) [reply groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/completion/inference/TypeInferenceVisitor.java:103]: I don't think so. In the time the metadata are null in this case.

File: groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/occurrences/TypeVisitor.java (hunk 2 above)
Comment (ppisl) [reply groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/occurrences/TypeVisitor.java:145]: For this moment yes.

**Pull request merged**